.github/workflows/ci.yml

```diff
@@ -40,18 +40,31 @@ jobs:
   unit-tests:
     name: Unit tests
     needs: [ pre-commit, towncrier, package ]
-    runs-on: ${{ matrix.platform }}-latest
+    runs-on: ${{ matrix.platform }}
     continue-on-error: ${{ matrix.experimental }}
     strategy:
       fail-fast: false
       matrix:
-        platform: [ "macos", "ubuntu", "windows" ]
-        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12", "3.13-dev" ]
+        platform: [ macos-12, macos-14, windows-latest, ubuntu-latest ]
```

Comment:
I'm not sure removing the quotes here (and below) is a good idea. YAML has an... interesting relationship with bare values; I've found it safer to unilaterally quote when you know the value is a string, even if it isn't *strictly* necessary (and it clearly isn't here, because CI is passing).

Reply:
This is definitely the thing I hate most about YAML....other than everything else. To add to this, I've also wondered how the GitHub substitutions are handled, i.e. `${{ xxx }}`, and whether any of these substitutions happen before a YAML processor runs...